scipy/special/_lebedev.py

```diff
@@ -0,0 +1,5453 @@
+
+            leb.x[start + 17] = -c
+            leb.y[start + 17] = a
+            leb.z[start + 17] = b
+            leb.w[start + 17] = 4.0 * pi * v
+
+            leb.x[start + 18] = c
+            leb.y[start + 18] = -a
+            leb.z[start + 18] = b
+            leb.w[start + 18] = 4.0 * pi * v
+
+            leb.x[start + 19] = c
+            leb.y[start + 19] = a
+            leb.z[start + 19] = -b
+            leb.w[start + 19] = 4.0 * pi * v
+
+            leb.x[start + 20] = -c
+            leb.y[start + 20] = -a
+            leb.z[start + 20] = b
+            leb.w[start + 20] = 4.0 * pi * v
+
+            leb.x[start + 21] = c
+            leb.y[start + 21] = -a
+            leb.z[start + 21] = -b
+            leb.w[start + 21] = 4.0 * pi * v
+
+            leb.x[start + 22] = -c
+            leb.y[start + 22] = a
+            leb.z[start + 22] = -b
+            leb.w[start + 22] = 4.0 * pi * v
+
+            leb.x[start + 23] = -c
+            leb.y[start + 23] = -a
+            leb.z[start + 23] = -b
+            leb.w[start + 23] = 4.0 * pi * v
+
+            leb.x[start + 24] = c
+            leb.y[start + 24] = b
+            leb.z[start + 24] = a
+            leb.w[start + 24] = 4.0 * pi * v
+
+            leb.x[start + 25] = -c
+            leb.y[start + 25] = b
+            leb.z[start + 25] = a
+            leb.w[start + 25] = 4.0 * pi * v
+
+            leb.x[start + 26] = c
+            leb.y[start + 26] = -b
+            leb.z[start + 26] = a
+            leb.w[start + 26] = 4.0 * pi * v
+
+            leb.x[start + 27] = c
+            leb.y[start + 27] = b
+            leb.z[start + 27] = -a
+            leb.w[start + 27] = 4.0 * pi * v
+
+            leb.x[start + 28] = -c
+            leb.y[start + 28] = -b
+            leb.z[start + 28] = a
+            leb.w[start + 28] = 4.0 * pi * v
+
+            leb.x[start + 29] = c
+            leb.y[start + 29] = -b
+            leb.z[start + 29] = -a
+            leb.w[start + 29] = 4.0 * pi * v
+
+            leb.x[start + 30] = -c
+            leb.y[start + 30] = b
+            leb.z[start + 30] = -a
+            leb.w[start + 30] = 4.0 * pi * v
+
+            leb.x[start + 31] = -c
+            leb.y[start + 31] = -b
+            leb.z[start + 31] = -a
+            leb.w[start + 31] = 4.0 * pi * v
+
+            leb.x[start + 32] = a
+            leb.y[start + 32] = c
+            leb.z[start + 32] = b
+            leb.w[start + 32] = 4.0 * pi * v
+
+            leb.x[start + 33] = -a
+            leb.y[start + 33] = c
+            leb.z[start + 33] = b
+            leb.w[start + 33] = 4.0 * pi * v
+
+            leb.x[start + 34] = a
+            leb.y[start + 34] = -c
+            leb.z[start + 34] = b
+            leb.w[start + 34] = 4.0 * pi * v
+
+            leb.x[start + 35] = a
+            leb.y[start + 35] = c
+            leb.z[start + 35] = -b
+            leb.w[start + 35] = 4.0 * pi * v
+
+            leb.x[start + 36] = -a
+   
```

Comment:
I wonder if it would be more natural from Python’s perspective to return a 2D array of size m-by-3 instead?

Reply:
This will be up to the `special` maintainers in case there is precedent to follow. But I typically prefer the axis representing the dimensionality of the space to come first because I can unpack it like:
```python3
x, y, z = x
# or just use, say x[0]
```

Reply:
Indeed, both choices have pros and cons. From a performance point of view (with respect to memory locality), certain operations, such as conversion from Cartesian to spherical coordinates, should perform better if the coordinates of a single point are placed next to each other in memory.

Of course, for this particular PR, the maximum possible data is just a few thousand floats. So either choice should have negligible performance difference.

Reply:
I don't think that's what I'd expect, and that's not what I observe.

<details>
Is this fair?

```python3
import numpy as np
rng = np.random.default_rng(23498234982345)

def p2c(polar):
    r, th = polar
    x = r * np.cos(th)
    y = r * np.sin(th)
    return x, y

def pT2c(polarT):
    r, th = polarT[:, 0], polarT[:, 1]
    x = r * np.cos(th)
    y = r * np.sin(th)
    return x, y

def c2p(cartesian):
    x, y = cartesian
    # r = np.linalg.vector_norm(cartesian, axis=1) makes only a slight difference
    # perhaps that's what you had in mind?
    r = (x*x + y*y)**0.5
    th = np.arctan2(y, x)
    return r, th

def cT2p(cartesianT):
    x, y = cartesianT[:, 0], cartesianT[:, 1]
    r = (x*x + y*y)**0.5
    th = np.arctan2(y, x)
    return r, th

polar = rng.random(size=(2, 10000))
polarT = polar.T.copy()
cartesian = polar.copy()
cartesianT = polarT.copy()
```

```
%timeit p2c(polar)
120 μs ± 15.7 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
%timeit pT2c(polarT)
106 μs ± 8.22 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
%timeit p2c(polar)
100 μs ± 6.64 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
%timeit pT2c(polarT)
105 μs ± 7.16 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
%timeit c2p(cartesian)
229 μs ± 22.8 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
%timeit c2p(cartesian)
225 μs ± 27 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
%timeit cT2p(cartesianT)
228 μs ± 15.3 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
%timeit cT2p(cartesianT)
241 μs ± 24.9 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

For larger of arrays, similar lack of clear trend, unless `p2c` and `c2p` are marginally faster.
</details>

In any case, speed wasn't the reason, and I think precedent within `special` is probably more important than opinions unless there's a clear winner.